### PR TITLE
ENH: resample_data="data" for LabelsMasker

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
Tentative fix for #340 

For now, it is only implemented for NiftiLabelsMasker. I will work on NiftiMapsMasker, but I welcome feedback eg from @AlexandreAbraham  and @lesteve

### CODE REVIEW

File: nilearn/_utils/niimg_conversions.py [hunk 1]
```diff
@@ -49,6 +49,16 @@ def _get_shape(img):
...
 
 
+def _check_same_fov(img1, img2):
+    """ Return True if img1 and img2 have the same field of view
+        (shape and affine), False elsewhere.
+    """
+    img1 = check_niimgs(img1, accept_3d=True)
+    img2 = check_niimgs(img2, accept_3d=True)
+    return (_get_shape(img1)[:3] == _get_shape(img2)[:3]
+            and np.allclose(img1.affine, img2.affine))
```
Comment (AlexandreAbraham) [inline nilearn/_utils/niimg_conversions.py]: Use `get_affine()`.

File: nilearn/input_data/nifti_region.py [hunk 2]
```diff
@@ -232,12 +230,26 @@ def transform(self, imgs, confounds=None):
...
 
-        if self.resampling_target == "labels":
+        if self.resampling_target == "data":
+            if not hasattr(self, '_resampled_labels_img_'):
+                self._resampled_labels_img_ = self.labels_img_
+            if not _check_same_fov(imgs, self._resampled_labels_img_):
+                logger.log("resampling labels", verbose=self.verbose)
+                self._resampled_labels_img_ = self._cache(image.resample_img,
+                    memory_level=1)(
+                        self.labels_img_, interpolation="continuous",
```
Comment (AlexandreAbraham) [inline nilearn/input_data/nifti_region.py]: `interpolation="nearest"`

File: nilearn/input_data/tests/test_nifti_region.py [hunk 3]
```diff
@@ -199,6 +200,22 @@ def test_nifti_labels_masker_2():
...
+    fmri22_img, _ = generate_random_img(shape22, affine=affine2,
+                                                 length=length)
+    masker = NiftiLabelsMasker(labels33_img, mask_img=mask22_img)
+
+    masker.fit_transform(fmri22_img)
+    np.testing.assert_array_equal(
+        masker._resampled_labels_img_.get_affine(),
+        affine2)
+
+
```
Comment (lesteve) [inline nilearn/input_data/tests/test_nifti_region.py]: Do we want a test that uses an explicit resampling_target="data" to check that it is the same as not specifying resampling_target?

Comment (GaelVaroquaux) [inline nilearn/input_data/tests/test_nifti_region.py]: I think that it is not necessary, as long as we keep an eye on the coverage.

**Comment by GaelVaroquaux** [comment]
Any more reviews, I think this is mergeable.

**Comment by lesteve** [comment]
LGTM

**Comment by GaelVaroquaux** [comment]
Tests pass. I am merging.

**Pull request merged**